src/ppx_deriving_yojson.ml

```diff
@@ -897,30 +871,90 @@ let desu_core_expr_of_typ typ =
-    ()
-  ))
+let make_gen f =
+  let f' ~ctxt x strict meta exn =
+    let is_strict = match strict with
+      | Some strict -> strict
+      | None -> true (* by default *)
+    in
+    let want_meta = match meta with
+      | Some meta -> meta
+      | None -> false (* by default *)
+    in
+    let want_exn = match exn with
+      | Some exn -> exn
+      | None -> false (* by default *)
+    in
+    let options = { is_strict; want_meta; want_exn } in
+    let path =
+      let code_path = Expansion_context.Deriver.code_path ctxt in
+      (* Cannot use main_module_name from code_path because that contains .cppo suffix (via line directives), so it's actually not the module name. *)
+      (* Ppx_deriving.module_from_input_name ported to ppxlib. *)
+      let main_module_path = match Expansion_context.Deriver.input_name ctxt with
+        | ""
+        | "_none_" -> []
+        | input_name ->
+          match Filename.chop_suffix input_name ".ml" with
+          | exception _ ->
+            (* see https://github.com/ocaml-ppx/ppx_deriving/pull/196 *)
+            []
+          | path ->
+            [String.capitalize_ascii (Filename.basename path)]
+      in
+      main_module_path @ Code_path.submodule_path code_path
+    in
+    f ~options ~path x
+  in
+  Deriving.Generator.V2.make (args ()) f'
+
+let _to_deriving: Deriving.t =
+  Deriving.add
+    "to_yojson"
+    ~str_type_decl:(make_gen (fun ~options ~path (_, type_decls) ->
+        structure (on_str_decls str_of_type_to_yojson) ~options ~path type_decls
+      ))
+    ~sig_type_decl:(make_gen (fun ~options ~path (_, type_decls) ->
+        on_sig_decls sig_of_type_to_yojson ~options ~path type_decls
+      ))
+    ~str_type_ext:(make_gen ser_str_of_type_ext)
+    ~sig_type_ext:(make_gen ser_sig_of_type_ext)
+
+let _of_deriving: Deriving.t =
+  Deriving.add
+    "of_yojson"
+    ~str_type_decl:(make_gen (fun ~options ~path (_, type_decls) ->
+        structure (on_str_decls str_of_type_of_yojson) ~options ~path type_decls
+      ))
+    ~sig_type_decl:(make_gen (fun ~options ~path (_, type_decls) ->
+        on_sig_decls sig_of_type_of_yojson ~options ~path type_decls
+      ))
+    ~str_type_ext:(make_gen desu_str_of_type_ext)
+    ~sig_type_ext:(make_gen desu_sig_of_type_ext)
+
+(* Not just alias because yojson also has meta (without its own deriver name) *)
+let _deriving: Deriving.t =
+  Deriving.add
+    "yojson"
+    ~str_type_decl:(make_gen (fun ~options ~path (_, type_decls) ->
+        structure (on_str_decls str_of_type) ~options ~path type_decls
+      ))
+    ~sig_type_decl:(make_gen (fun ~options ~path (_, type_decls) ->
+        on_sig_decls sig_of_type ~options ~path type_decls
+      ))
+    ~str_type_ext:(make_gen str_of_type_ext)
+    ~sig_type_ext:(make_gen sig_of_type_ext)
+
+(* custom extensions such that "derive"-prefixed also works *)
+let to_derive_extension =
+  Extension.V3.declare "derive.to_yojson" Extension.Context.expression
```

Comment:
I think it's good practice to add the name of the ppx as the prefix "root", i.e. here to register those extensions as `ppx_deriving_yojson.derive.to_yojson`. It's unlikely that users will ever have to go that far as to use the full name but that way we're pretty sure not to have a conflict here.